src/js/components/DataTable/Body.js

```diff
@@ -43,177 +43,198 @@ const Row = memo(
-              size: 'auto',
-              render: () => (
-                <CheckBox
-                  tabIndex={onClickRow === 'select' ? -1 : undefined}
-                  a11yTitle={`${
-                    isSelected ? 'unselect' : 'select'
-                  } ${primaryValue}`}
-                  checked={isSelected}
-                  disabled={isDisabled || !onSelect}
-                  onChange={() => {
-                    if (isSelected) {
-                      onSelect(
-                        selected.filter((s) => s !== primaryValue),
-                        datum,
-                      );
-                    } else onSelect([...selected, primaryValue], datum);
-                  }}
-                  pad={cellProps.pad}
-                />
-              ),
-            }}
-            verticalAlign={verticalAlign}
-          />
-        )}
-        {rowDetails && (
-          <ExpanderCell
-            background={isSelected && cellProps.selected.background}
-            context={isRowExpanded ? 'groupHeader' : 'body'}
-            expanded={isRowExpanded}
-            expandLabel={expandLabel}
-            onToggle={() => {
-              let nextRowExpand;
-              const rowKey = primaryValue || index;
-              if (isRowExpanded) {
-                nextRowExpand = rowExpand.filter((s) => s !== rowKey);
-              } else {
-                nextRowExpand = [...rowExpand, rowKey];
-              }
-              if (rowDetails.onExpand) {
-                rowDetails.onExpand(nextRowExpand, datum);
-              } else {
-                setRowExpand(nextRowExpand);
+          }}
+          onMouseOver={
+            onClickRow && !isDisabled ? () => setActive(index) : undefined
+          }
+          tabIndex={
+            // eslint-disable-next-line no-nested-ternary
+            !onClickRow
+              ? undefined
+              : // If this row is focused, it should be focusable
+              // If no row is focused and this is the first row,
+              // make it focusable
+              (focused !== undefined && focused === index) ||
+                (focused === undefined && lastFocused === index) ||
+                (focused === undefined && index === 0)
+              ? 0
+              : // Otherwise, not in tab order
+                -1
+          }
+          {...passThemeFlag}
+        >
+          {(selected || onSelect) && (
+            <Cell
+              background={
+                (isSelected && cellProps.selected.background) ||
+                (pinnedOffset?._grommetDataTableSelect &&
+                  cellProps.pinned.background) ||
+                cellProps.background
               }
-            }}
-            pad={cellProps.pad}
-            verticalAlign={verticalAlign}
-          />
-        )}
-        {columns.map((column) => (
-          <Cell
-            key={column.property}
-            background={
+              // should this background also be the same ??
```

Comment:
this background is this part of the DataTable, should we also account for the background here?

@taysea @jcfilben 

<img width="195" height="352" alt="Screenshot 2025-08-18 at 10 03 09 PM" src="https://github.com/user-attachments/assets/2c41ad01-c998-4037-b561-3b695a73ba0c" />

Reply:
okay thats fine just wanted to confirm so Ill change that logic as well